scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala

```diff
@@ -449,6 +455,14 @@ class BigQueryClient private (private val projectId: String,
     }
   }
 
+  private def getUdfResource(): Option[java.util.List[UserDefinedFunctionResource]] = {
+    BigQueryClient.udfResourceUri.map { uri =>
```

Comment:
Not sure about the `sys.props` approach though since it can't handle different resources per query or multiple resources.

To make it work with `fromQuery` we might have to create a new annotation?
Also I wonder how many users actually need this vs doing UDF in native Scio code?
Anyway I'd defer this until we have a better idea.